cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/BasicDatastore.java

```diff
@@ -643,7 +643,7 @@ public LocalDocument call(SQLDatabase db) throws Exception {
         return null;
     }
 
-    private BasicDocumentRevision createDocument(SQLDatabase db,String docId, final DocumentBody body)
+    private BasicDocumentRevision createJsonDocument(SQLDatabase db, String docId, final DocumentBody body)
```

Comment:
Candidates I would suggest are `insertRevision` `insertRevisionBody` `insertRevisionJson` - something which emphasises the fact that it calls `insertRevision` and serialises the `DocumentBody` into JSON bytes before doing so.